cmake/EthCompilerSettings.cmake

```diff
@@ -104,6 +104,10 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MA
 			# Abort if linking results in any undefined symbols
 			# Note: this is on by default in the CMake Emscripten module which we aren't using
 			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s ERROR_ON_UNDEFINED_SYMBOLS=1")
+			# Define additional methods to export in soljson.js. Needed by solc-js.
+			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s EXTRA_EXPORTED_RUNTIME_METHODS=['cwrap','addFunction','removeFunction','Pointer_stringify','lengthBytesUTF8','_malloc','stringToUTF8','setValue']")
+			# Do not build as a WebAssembly target - we need an asm.js output.
+			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s WASM=0")
```

Comment:
Why do we need asm.js output and drop explicitly WASM output? (just curious for my own understanding).

Reply:
I think the plan for WASM is to go directly through LLVM and not to go through Emscripten. We did look at what Emscripten outputs, when requesting WASM, but there were quite some issues with it - also solc-js would need to be adjusted to work with a WASM build. So out of scope here. Emscripten enabled it by default in newer versions, though, so we have to disable it here.